src/lib/db/query.js

```diff
@@ -98,7 +98,25 @@ module.exports = {
       data => data.count
     )
     .catch(e => {
-      global.logger.error(`Problem retrieving a search from "${from}" to "${to}" for group "${group}"`, e);
+      global.logger.error(`Problem retrieving a search from "${from}" to "${to || 'now'}" for group "${group}"`, e);
+      throw new Error('Could not fetch data');
+    }),
+
+  hourlyUsage: (from, to, group) => db.manyOrNone(
+    sqlBuilder({
+      'SELECT': 'COUNT(*)::INTEGER, weekend::INTEGER, hour',
+      'FROM': '(',
+      '  SELECT': 'TO_CHAR(date_time AT TIME ZONE \'europe/london\', \'HH24\')::INTEGER AS hour,\n' +
+      '    TO_CHAR(date_time AT TIME ZONE \'europe/london\', \'DAY\') LIKE \'S%\' AS weekend',
+      '  FROM': 'lev_audit',
+      '  WHERE': [from && fromDate, to && toDate,
+        group ? searchGroup : 'groups::TEXT NOT ILIKE \'%monitor%\' AND client = \'lev-web\''],
+      ')': 'AS counts',
+      'GROUP BY': 'weekend, hour'
```

Comment:
Isn't this conflating two different things?

I'd think we want two queries, one grouping by hour of the day and one grouping by day of the week.

Reply:
We already have a chart for daily usage. The intention here is to separate the weekends and weekdays, because they have such wildly different usage.

Reply:
But do they? Most of the time, the weekend usage by hour is going to look like the weekday usage by hour only the line will be lower.

(I'll explain more about my over reservations in another comment.)

> We already have a chart for daily usage.
But we don't have one specifically broken down by day of the week. (Mo, Tu, We, Th, Fr, Sa, Su) It's true that you can see the weekends usually but they aren't labelled explicitly.